suse/x86_64/suse-leap-42.3-JeOS/root/usr/lib/systemd/system/data_part.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Add a data partition
+ConditionPathExists=/.kiwi_data_part.trigger
```

Comment:
I agree with all you have said regarding the initrd and trying to keep it as simple as possible. Initially I thought about a dracut module to avoid something like this ConditionPathExists, which can be tricky in a readonly filesystem. That's why I thought about a reproducing oem-repart behavior.

Anyway, probably we could think of a service that it is not a oneshot service which runs at every boot and just checks if the partition is there at the very beginning, like oem-repart does. This way we would not have to relay on writing files on into the root filesystem.

Reply:
ah yes very good point. the caasp thing uses a read-only root. Let me address this scenario. I'll come up with a commit next. Thanks